pkg/clusterconfig/clusterconfig.go

```diff
@@ -44,10 +48,19 @@ type S3 struct {
 	Region    string
 }
 
+type Swift struct {
+	// UPI crednetials
+	Username string
+	Password string
+	// IPI auth data from clouds.yaml file
+	Cloud *clientconfig.Cloud
```

Comment:
I'm not keen on storing the full `clouds.yaml` content here, especially if credentials are reused.
I'd rather all the data needed to create the openstack driver be added here as separate fields.

See the s3 config struct [1] - we only store what we need, and read in data provided by the cred minter for IPI.
[1] https://github.com/openshift/cluster-image-registry-operator/blob/master/pkg/clusterconfig/clusterconfig.go#L40-L45

Reply:
Done. I added all the required fields to the struct.